src/Core/Content/Product/SearchKeyword/ProductSearchBuilder.php

```diff
@@ -71,8 +95,51 @@ public function build(Request $request, Criteria $criteria, SalesChannelContext
         );
     }
 
-    public function buildParent(Request $request, Criteria $criteria, SalesChannelContext $context): void
+
+    public function buildShopware64AndGreater(Request $request, Criteria $criteria, SalesChannelContext $context): void
     {
-        parent::build($request, $criteria, $context);
+        $search = $request->query->get('search');
+
+        if (is_array($search)) {
+            $term = implode(' ', $search);
+        } else {
+            $term = (string) $search;
+        }
+
+        $term = trim($term);
+
+        $pattern = $this->interpreter->interpret($term, $context->getContext());
+
+        foreach ($pattern->getTerms() as $searchTerm) {
+            $criteria->addQuery(
+                new ScoreQuery(
+                    new EqualsFilter('product.searchKeywords.keyword', $searchTerm->getTerm()),
+                    $searchTerm->getScore(),
+                    'product.searchKeywords.ranking'
+                )
+            );
+        }
+        $criteria->addQuery(
+            new ScoreQuery(
+                new ContainsFilter('product.searchKeywords.keyword', $pattern->getOriginal()->getTerm()),
+                $pattern->getOriginal()->getScore(),
+                'product.searchKeywords.ranking'
+            )
+        );
```

Comment:
This was intended, as the methods are copied from SW:
* https://github.com/shopware/platform/blob/trunk/src%2FCore%2FContent%2FProduct%2FSearchKeyword%2FProductSearchBuilder.php

We need to overrite it, to get rid of the `MissingRequestParameterException` which prevents empty search. In case SW does some modifications again, we just need to add another method for the version and don't need to debug the combined methods.